Add OMO:0003014 "has ontology hierarchical property"

cthoyt opened the pull request:
Closes https://github.com/information-artifact-ontology/IAO/issues/292

In the same way that you can use IAO:0000700 (has ontology root term) to annotate onto an ontology that it has root terms, you can use OMO:0003014 to annotate onto an ontology that it uses the given predicates as its hierarchy.

Here's a real world example using the Developing Mouse Brain Atlas Ontology (DBMA), which uses both subclass and part-of relationships in its hierarchy. I didn't make this example up, this is in the EBI's OLS4 configuration:

```turtle
<https://purl.brain-bican.org/ontology/dmbao/dmbao.owl> a owl:Ontology; 
    OMO:0003014 rdfs:subClassOf, BFO:0000050 .
```

cthoyt committed:
Add OMO:0003014 "has ontology hierarchical property"

Closes https://github.com/information-artifact-ontology/IAO/issues/292

cthoyt commented:
@matentzn

matentzn commented on src/templates/annotation_properties.tsv:
```diff
@@ -16,3 +16,4 @@ OMO:0003012	owl:AnnotationProperty	acronym	A synonym type for describing abbrevi
 oboInOwl:hasDbXref	owl:AnnotationProperty	has cross-reference	An annotation property that links an ontology entity or a statement to a prefixed identifier or URI.		https://orcid.org/0000-0002-7356-1779	2024-03-18	disease characteristic (MONDO:0021125) has cross-reference (http://www.geneontology.org/formats/oboInOwl#hasDbXref) "NCIT:C41009"^^xsd:string	https://github.com/information-artifact-onto ...
 OMO:0004000	owl:AnnotationProperty	has symbol	An information content entity that is a mark(s) or character(s) used as a conventional representation of another entity.		https://orcid.org/0000-0002-6523-4866	2024-03-25	A serial number such as "12324X"; a stop sign; a written proper name such as "OBI"	https://github.com/information-artifact-ontology/ontology-metadata/issues/144
 OMO:0002001	owl:AnnotationProperty	preferred label for community	A provenance property that denotes a synonym to be the preferred one for a specific community.		https://orcid.org/0000-0002-7356-1779	2025-06-02	The property is intended to be used exclusively on synonym assertions, to express that the synonym in question is the preference of a specific community. For example, one community may pref ...
+OMO:0003014	owl:AnnotationProperty	has ontology hierarchical property	Relates an ontology to a predicates that is used to construct its hierarchy. Typically, this is rdfs:subClassOf, but in ontologies that are partonomies, this might also include BFO:0000050 (part of). Display tools like OLS can use predicates annotated with this property for rendering the ontology class hierarchy. There can be m ...
```
```suggestion
OMO:0003014	owl:AnnotationProperty	has ontology hierarchical property	Relates an ontology to a predicate that is used to construct its hierarchy. Typically, this is rdfs:subClassOf, but in ontologies that are partonomies, this might also include BFO:0000050 (part of). Display tools like OLS can use predicates annotated with this property for rendering the ontology class hierarchy. There can be more than one hierarchical property. In the absence of any, it should be assumed that it is rdfs:subClassOf.		https://orcid.org/0000-0003-4423-4370	2025-09-17	dmba uses BFO:0000050 as a hierarchical property.	https://github.com/information-artifact-ontology/IAO/issues/292

```

matentzn reviewed (changes requested):
We need some really clear picture and on how this is supposed to be used. if this is for hierarchical UI tools, so basically the info that OLS needs, we need to see if it is even POSSIBLE to say:

```
my:ont a owl:Ontology; OMO:0003014 rdfs:subClassOf.
```
Assuming this is how you want to use it - we also need a stronger example here and a clearer use case because we cant be using a property like this for anything OTHER than instructing UI tools..

cthoyt commented:
@matentzn yes, your example is exactly what I am thinking. I wrote a concrete one in the issue description and will also put it in the PR.

Am I allowed to annotate domain/range constraints onto this to make this more explicit?

cthoyt committed:
Update annotation_properties.tsv

matentzn commented:
Did you try to make such an annotation in an ontology (esp. with subclass of)? I would like to know if this would cause an OWL2 profile violation https://robot.obolibrary.org/validate-profile.

cthoyt commented:
I tried it in OMO making a different dummy annotation property and it didn't seem to make an issue (though it appears OMO has other violations that are unrelated)

cthoyt commented:
@matentzn I created a fully minimal example that shows this isn't an issue in https://gist.github.com/cthoyt/427b8b00294b75f3135e9495193991f6

matentzn committed:
Rephrased the definition to make intended usage even more explicit

matentzn committed:
Add 'has ontology hierarchical property' annotation

Introduces OMO_0003014 annotation property to specify predicates for rendering ontology hierarchies in user interfaces. Includes label, definition, usage example, contributor, creation date, and related issue reference.

matentzn commented:
I rephrased the intended usage slightly (without changing the meaning). Can you check you are ok? then I will find more reviewers.

cthoyt commented:
@matentzn yes, I am happy with your updates, thanks!

matentzn commented:
@zhengj2007 this PR is ready for your review. Thank you!
